main.py

```diff
@@ -129,27 +135,30 @@ def app_startup():
         settings.pretrained_model_name_or_path, cuda_device=settings.cuda_device
     )
 
-    with open(settings.index_filepath, "r") as f:
-        text = f.readlines()
-    embeddings = _index(text)
-    index.text = text
-    index.embeddings = embeddings
-
 
-@app.post("/query/")
+@app.post("/")
 async def query(query: Query):
     embedded_query = _encode(
-        [query.text], model.tokenizer, model.model, mean_pool=settings.mean_pool
+        [query.query.text], model.tokenizer, model.model, mean_pool=settings.mean_pool
     )
 
+    text = [document.text for document in query.documents]
+    ids = [document.uid for document in query.documents]
+    embeddings = _index(text)
+    index.text = text
+    index.ids = ids
+    index.embeddings = embeddings
+
     cos = torch.nn.CosineSimilarity(-1)
     similarity_scores = cos(embedded_query, index.embeddings)
-    top_k_scores, top_k_indicies = torch.topk(similarity_scores, query.top_k)
+    # If top_k not specified, return all documents.
+    top_k = query.top_k or similarity_scores.size(0)
```

Comment:
Needs to be bounded and sanitized (falsy when query.top_k is 0)

Reply:
Good catch, thanks